src/battle_script_commands.c

```diff
@@ -3958,6 +3961,18 @@ static void Cmd_getexp(void)
     case 6: // increment instruction
         if (gBattleControllerExecFlags == 0)
         {
+        #if B_RECALCULATE_STATS >= GEN_5
+            // Recalculate the stats of every party member before the end
+            for (i = 0; i < PARTY_SIZE; i++)
+            {
+                if (GetMonData(&gPlayerParty[i], MON_DATA_SPECIES2) != SPECIES_NONE
+                 && GetMonData(&gPlayerParty[i], MON_DATA_SPECIES2) != SPECIES_EGG)
+                {
+                    CalculateMonStats(&gPlayerParty[i]);
```

Comment:
This should be moved to the battle's end, not after receiving exp.

Reply:
`HandleEndTurn_FinishBattle`?